src/Tizen.NUI/src/internal/Application.cs

```diff
@@ -1194,12 +1195,22 @@ public static Application New(int argc, string stylesheet, Application.WindowMod
 
         public static Application New(string[] args, string stylesheet, Application.WindowMode windowMode)
         {
-            int argc = args.Length;
-            string argvStr = string.Join(" ", args);
-
-            Application ret = new Application(NDalicPINVOKE.Application_New__MANUAL_4(argc, argvStr, stylesheet, (int)windowMode), true);
+            Application ret = null;
+            try
+            {
+                int argc = args.Length;
+                string argvStr = string.Join(" ", args);
 
-            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+                ret = new Application(NDalicPINVOKE.Application_New__MANUAL_4(argc, argvStr, stylesheet, (int)windowMode), true);
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
```

Comment:
I think `NDalicPINVOKE.SWIGPendingException.Pending` and `NDalicPINVOKE.SWIGPendingException.Retrieve()` should be an atomic operation.

Reply:
https://github.com/Samsung/TizenFX/blob/d937dc176cc23d88355b9133867a6210d121dc76/src/Tizen.NUI/src/internal/Interop/NDalicPINVOKE.cs#L181

Currently, Retrieve and Set which access numExceptionsPending value are synchronized to Lock.
Is your comment that you need lock even when you get it from Pending?

Reply:
Simply, imagine two threads calling like so;
```
Thread1: `pending`------------------------------->'Retrieve()'
Thread2: -------------'pending'->'Retrieve()'---------------->
```

Reply:
I figured https://github.com/Samsung/TizenFX/blob/d937dc176cc23d88355b9133867a6210d121dc76/src/Tizen.NUI/src/internal/Interop/NDalicPINVOKE.cs#L151 so it might not be the case I depicted above.

Yet, it's hard to figure out the use of `numExceptionPending` nor the meaning of  API separation of for checking and retrieving exception object.